rpc/block_svc_routines.c

```diff
@@ -258,6 +260,67 @@ blockStr2arrayAddToJsonObj(json_object *json_obj, char *string, char *label)
 }
 
 
+static int
+blockCheckBlockLoadedStatus(char *block_name, char *gbid, blockResponse *reply)
+{
+
+  int ret = -1;
+  char *exec = NULL;
+  int is_loaded = true;
+
+
+  if (GB_ASPRINTF(&exec, GB_TGCLI_CHECK, block_name, gbid) == -1) {
+    goto out;
+  }
+
+  ret = gbRunner(exec);
+  if (ret == -1) {
+    GB_ASPRINTF(&reply->out, "command exit abnormally for %s\n", block_name);
+    LOG("mgmt", GB_LOG_ERROR, "%s", reply->out);
+    goto out;
+  } else if (ret == 1) {
+    is_loaded = false;
+    GB_ASPRINTF(&reply->out, "Block '%s' may be not loaded\n", block_name);
+    LOG("mgmt", GB_LOG_ERROR, "%s", reply->out);
+  }
+  GB_FREE(exec);
+
+  if (GB_ASPRINTF(&exec, GB_SAVECONFIG_CHECK, block_name) == -1) {
+    goto out;
+  }
+
+  ret = gbRunner(exec);
+  if (ret == -1) {
+    GB_FREE(reply->out);
+    GB_ASPRINTF(&reply->out, "command exit abnormally for %s\n", block_name);
+    LOG("mgmt", GB_LOG_ERROR, "%s", reply->out);
+    goto out;
+  } else if (ret == 1 || ret == 2) {  /* search not found or savefile not exist */
```

Comment:
same comment as above

Reply:
Yes, 2 is extra here for file not found case (grep on savefile case)

Reply:
Isn't it better to flip the condition? If ret == 0 consider success otherwise consider failure? Isn't that more robust?

Reply:
will generalize this part later, may be in a different revision as this is well tested lets live with it for now :-)